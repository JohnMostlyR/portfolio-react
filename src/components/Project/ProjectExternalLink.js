import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import typography from '../../styles/templates/typography';
import {visuallyHidden} from '../../styles/mixins';

const ListItem = styled.li`
  display: inline-block;
  margin-right: 1vw;
  transition: transform .2s ease-in-out;
  vertical-align: middle;

  &:hover {
    transform: scale(1.5);
  }
`;

const Link = styled.a`
  color: #fff;
  ${typography.doublePica}
`;

const LinkDescription = styled.span`
  ${visuallyHidden}
`;

const ProjectExternalLink = (props) => (
  <ListItem data-tip={props.name}>
    <Link
      href={props.url}
      target="_blank"
      rel="noopener noreferrer">
      <span className={props.icon} aria-hidden="true"/>
      <LinkDescription>{props.name}</LinkDescription>
    </Link>
  </ListItem>
);

ProjectExternalLink.propTypes = {
  icon: PropTypes.string,
  name: PropTypes.string,
  url: PropTypes.string,
};

ProjectExternalLink.defaultPropTypes = {
  name: 'Name',
  url: '#',
};

export default ProjectExternalLink;
