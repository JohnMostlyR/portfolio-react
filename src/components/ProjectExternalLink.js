import React from 'react';

const ProjectExternalLink = (props) => (
  <li className="c-portfolio-project-links-list__item">
    <a
      className="c-portfolio-project-links-list__link"
      href={props.link.url}
      target="_blank"
      rel="noopener noreferrer">
      <span className={props.link.icon} aria-hidden="true"/>
      <span className="u-visually-hidden">{props.link.name}</span>
    </a>
  </li>
);

export default ProjectExternalLink;
