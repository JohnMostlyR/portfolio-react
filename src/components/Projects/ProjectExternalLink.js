import React from 'react';
import styled from 'styled-components';
import typography from '../../styles/templates/typography';
import {visuallyHidden} from '../../styles/mixins';

const ListItem = styled.li`
  display: inline-block;
  margin-right: 1vw;
  transition: transform .2s ease-in-out;
  vertical-align: middle;

  &:hover {
    transform: scale(1.5);
  }
`;

const Link = styled.a`
  color: #fff;
  ${typography.doublePica}
`;

const LinkDescription = styled.span`
  ${visuallyHidden}
`;

const ProjectExternalLink = (props) => (
  <ListItem data-tip={props.link.name}>
    <Link
      href={props.link.url}
      target="_blank"
      rel="noopener noreferrer">
      <span className={props.link.icon} aria-hidden="true"/>
      <LinkDescription>{props.link.name}</LinkDescription>
    </Link>
  </ListItem>
);

export default ProjectExternalLink;
