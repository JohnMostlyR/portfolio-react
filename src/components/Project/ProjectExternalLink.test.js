import React from 'react';
import {shallow} from 'enzyme';

import ProjectExternalLink from './ProjectExternalLink';

describe('ProjectExternalLink', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ProjectExternalLink/>);
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});