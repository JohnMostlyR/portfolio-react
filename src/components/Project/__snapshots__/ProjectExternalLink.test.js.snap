// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectExternalLink should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectExternalLink
    faIcon="meh-o"
    name="Name"
    url="#"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.a
        href="#"
        rel="noopener noreferrer"
        target="_blank"
>
        <FontAwesome
                fixedWidth={true}
                name="meh-o"
        />
        <styled.span>
                Name
        </styled.span>
</styled.a>,
      "data-tip": "Name",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FontAwesome
            fixedWidth={true}
            name="meh-o"
/>,
          <styled.span>
            Name
</styled.span>,
        ],
        "href": "#",
        "rel": "noopener noreferrer",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "fixedWidth": true,
            "name": "meh-o",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Name",
          },
          "ref": null,
          "rendered": "Name",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.a
          href="#"
          rel="noopener noreferrer"
          target="_blank"
>
          <FontAwesome
                    fixedWidth={true}
                    name="meh-o"
          />
          <styled.span>
                    Name
          </styled.span>
</styled.a>,
        "data-tip": "Name",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <FontAwesome
              fixedWidth={true}
              name="meh-o"
/>,
            <styled.span>
              Name
</styled.span>,
          ],
          "href": "#",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "fixedWidth": true,
              "name": "meh-o",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Name",
            },
            "ref": null,
            "rendered": "Name",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
